server/routerlicious/packages/lambdas/src/scriptorium/lambda.ts

```diff
@@ -88,12 +97,21 @@ export class ScriptoriumLambda implements IPartitionLambda {
         Promise.all(allProcessed).then(
             () => {
                 this.current.clear();
+                if (this.telemetryEnabled) {
+                    metric.setProperty("processingComplete", true);
+                }
                 // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
                 this.context.checkpoint(batchOffset!);
```

Comment:
if `this.context.checkpoint(batchOffset!);` throws an exception in line 104, we might not complete this metric. That's because we only call `success()` and `error()` in lines 107 and 113, but I am afraid those would not be executed if an exception is thrown in line 104. I believe that is because the `(error) => { }` section in line 111 only applies to errors thrown from the `allProcessed` promises, and not errors that get thrown within the `.then()` block.

A solution might be to wrap this `.then()` in a try/catch block, and also add a `metric.error()` statement in the catch block.

Reply:
Thats a valid concern @hedasilv, however I thought the `(error)` block on line 111 would execute for any errors thrown either from the `allProcessed` promises, or from the `.then()` block. Example: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/catch#using_and_chaining_the_catch_method

But looking at it again now, it seems the `(error)` block is still inside `then` block.. so it might not catch those errors. I will verify this and update, thanks :)